packages/amplify-category-function/src/index.ts

```diff
@@ -138,7 +138,7 @@ export async function getInvoker(context: any, params: InvokerParameters): Promi
 export function isMockable(context: any, resourceName: string): IsMockableResponse {
   const { service, dependsOn } = context.amplify.getProjectMeta()[category][resourceName];
   const hasLayer =
-    service === ServiceName.LambdaFunction && dependsOn.filter(dependency => dependency.category === 'function').length !== 0;
+    service === ServiceName.LambdaFunction && dependsOn && dependsOn.filter(dependency => dependency.category === 'function').length !== 0;
```

Comment:
I don't think it should ever not be an array, but maybe also check Array.isArray(dependsOn)